src/osd/ECTransaction.cc

```diff
@@ -196,9 +196,6 @@ void ECTransaction::generate_transactions(
       xattr_rollback[ECUtil::get_hinfo_key()] = old_hinfo;
       
       if (op.is_none() && op.truncate && op.truncate->first == 0) {
-	assert(op.truncate->first == 0);
-	assert(op.truncate->first ==
-	       op.truncate->second);
```

Comment:
I’m not sure exactly what’s going on here, but we check for this equality below so I don’t think it’s redundant. (Maybe inaccurate?)

Did some other issue lead you here, or did you find it by code inspection? Did you look at the git history to see how it happened?